core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java

```diff
@@ -528,4 +537,39 @@ protected void prepareCreation(Environment resource) {
     public WorkspaceResource resource() {
         return ENVIRONMENT;
     }
+
+    private void createAndSetNetwork(Environment environment, EnvironmentNetworkV4Request networkRequest) {
+        CloudPlatform cloudPlatform = CloudPlatform.valueOf(environment.getCloudPlatform());
+        BaseNetwork network = createNetworkIfPossible(environment, networkRequest, cloudPlatform);
+        if (network != null) {
+            environment.setNetwork(network);
+        }
+    }
+
+    private BaseNetwork createNetworkIfPossible(Environment environment, EnvironmentNetworkV4Request networkRequest, CloudPlatform cloudPlatform) {
+        BaseNetwork network = null;
+        if (networkRequest != null) {
+            if (CloudPlatform.AWS.equals(cloudPlatform) && networkRequest.getAws() != null) {
```

Comment:
This if is very smelly. The same refactor with cloud provider specific network creators is required here.